benchmark/compare.js

```diff
@@ -48,29 +56,71 @@ for (const filename of benchmarks) {
   }
 }
 
+if (showProgress) {
+  console.error(`Scheduled ${queue.length} runs in total: `);
+  for (const filename of benchmarks) {
+    console.error(`  ${runsPerFile} runs: ` +
+                  `${runs} x { ${binaries} } x ${filename}`);
+  }
+}
+
+function style(obj) {
+  return util.inspect(obj, {colors: true, breakLength: 80});
+}
+
+function getSeconds(time) {
+  return time[0] + time[1] / 1e9;
+}
+
 // Print csv header
 console.log('"binary", "filename", "configuration", "rate", "time"');
 
 (function recursive(i) {
   const job = queue[i];
 
+  if (i % runsPerFile === 0) {
+    runStartTime = process.hrtime();
+  }
+
+  const childEnv = Object.assign({}, process.env);
+  if (showProgress) {
+    childEnv.NODE_RUN_BENCHMARK_PROGRESS = '';
+  }
+
   const child = fork(path.resolve(__dirname, job.filename), cli.optional.set, {
-    execPath: cli.optional[job.binary]
+    execPath: cli.optional[job.binary],
+    env: childEnv
   });
 
   child.on('message', function(data) {
-    // Construct configuration string, " A=a, B=b, ..."
-    let conf = '';
-    for (const key of Object.keys(data.conf)) {
-      conf += ' ' + key + '=' + JSON.stringify(data.conf[key]);
-    }
-    conf = conf.slice(1);
+    if (data.type === 'report') {
+      // Construct configuration string, " A=a, B=b, ..."
+      let conf = '';
+      for (const key of Object.keys(data.conf)) {
+        conf += ' ' + key + '=' + JSON.stringify(data.conf[key]);
+      }
+      conf = conf.slice(1);
+      // Escape quotes (") for correct csv formatting
+      conf = conf.replace(/"/g, '""');
 
-    // Escape quotes (") for correct csv formatting
-    conf = conf.replace(/"/g, '""');
-
-    console.log(`"${job.binary}", "${job.filename}", "${conf}", ` +
-                `${data.rate}, ${data.time}`);
+      console.log(`"${job.binary}", "${job.filename}", "${conf}", ` +
+                  `${data.rate}, ${data.time}`);
+    } else if (showProgress && data.type === 'config') {
```

Comment:
I find this code rather hard to read, I don't understand it so it is hard to be constructive but maybe:
* reduce the amount of global variables.
* explain `lines`, `runLines` and `allLines`.
* wrap complexity in a function.